rediscluster/mapping.go

```diff
@@ -57,6 +73,33 @@ func (c *Cluster) newNode(addr string, initial bool) (*node, error) {
 	return node, nil
 }
 
+func (c *Cluster) nodeOpts(addr string) (*redisconn.Opts, error) {
+	nodeOpts := c.opts.HostOpts
+
+	if !nodeOpts.TLSEnabled {
+		return &nodeOpts, nil
+	}
+
+	originalHost, err := redisclusterutil.GetHost(addr)
+	if err != nil {
+		return nil, err
```

Comment:
The problem here is that we return a naked error from `net.SplitHostPort`. As you may have noticed, `redispipe` is very careful when it comes to dealing with errors. I believe we should wrap the error here in a package-specific error type, even though it's almost impossible to have an error here.

Reply:
fixed